experimental is failing on `TestInspectNetwork`

https://jenkins.dockerproject.org/job/Docker-PRs-experimental/45926/consoleFull

```
01:03:29 --- FAIL: TestInspectNetwork (5.04s)
01:03:29     inspect_test.go:18: Creating a new daemon
01:03:29     daemon.go:336: [d31da1d0eee59] waiting for daemon to start
01:03:29     daemon.go:336: [d31da1d0eee59] waiting for daemon to start
01:03:29     daemon.go:336: [d31da1d0eee59] waiting for daemon to start
01:03:29     daemon.go:336: [d31da1d0eee59] waiting for daemon to start
```

Error messages:

```
time="2019-07-13T01:03:25.303653977Z" level=error msg="error reading the kernel parameter net.ipv4.neigh.default.gc_thresh2" error="open /proc/sys/net/ipv4/neigh/default/gc_thresh2: no such file or directory"
time="2019-07-13T01:03:25.303669202Z" level=error msg="error reading the kernel parameter net.ipv4.neigh.default.gc_thresh3" error="open /proc/sys/net/ipv4/neigh/default/gc_thresh3: no such file or directory"
time="2019-07-13T01:03:25.303685835Z" level=error msg="error reading the kernel parameter net.ipv4.neigh.default.gc_thresh1" error="open /proc/sys/net/ipv4/neigh/default/gc_thresh1: no such file or directory"
time="2019-07-13T01:03:25.365879415Z" level=error msg="error reading the kernel parameter net.ipv4.vs.expire_nodest_conn" error="open /proc/sys/net/ipv4/vs/expire_nodest_conn: no such file or directory"
time="2019-07-13T01:03:25.550470174Z" level=error msg="error reading the kernel parameter net.ipv4.vs.expire_nodest_conn" error="open /proc/sys/net/ipv4/vs/expire_nodest_conn: no such file or directory"
time="2019-07-13T01:03:27.471109224Z" level=error msg="Could not get ipvs family information from the kernel. It is possible that ipvs is not enabled in your kernel. Native loadbalancing will not work until this is fixed."
time="2019-07-13T01:03:28.609238397Z" level=error msg="Inconsistent driver and libnetwork state for endpoint 7247aa6ba3417af3500405cc16bd86795e19466a71cad964ef767d3b3cdba4e9"
time="2019-07-13T01:03:28.610509263Z" level=error msg="Inconsistent driver and libnetwork state for endpoint 7247aa6ba3417af3500405cc16bd86795e19466a71cad964ef767d3b3cdba4e9"
time="2019-07-13T01:03:28.612106921Z" level=error msg="Inconsistent driver and libnetwork state for endpoint 7247aa6ba3417af3500405cc16bd86795e19466a71cad964ef767d3b3cdba4e9"
time="2019-07-13T01:03:28.620044315Z" level=error msg="Inconsistent driver and libnetwork state for endpoint 7247aa6ba3417af3500405cc16bd86795e19466a71cad964ef767d3b3cdba4e9"
time="2019-07-13T01:03:28.630295745Z" level=error msg="Error getting tasks: service yblygwx0w3n4dvm4ptgqs6xyj not found"
time="2019-07-13T01:03:28.635662636Z" level=error msg="failed to remove node" error="rpc error: code = Aborted desc = dispatcher is stopped" method="(*Dispatcher).Session" node.id=3oz6vheefok39p6f0zmfhgsci node.session=j6o0l9cxk86424zxnt91sfzup
time="2019-07-13T01:03:28.635931450Z" level=error msg="failed to receive changes from store watch API" error="rpc error: code = Unknown desc = context canceled"
time="2019-07-13T01:03:28Z" level=error msg="set up rule failed, [-t mangle -D INPUT -d 10.0.0.2/32 -j MARK --set-mark 256]:  (iptables failed: iptables --wait -t mangle -D INPUT -d 10.0.0.2/32 -j MARK --set-mark 256: iptables: No chain/target/match by that name.\n (exit status 1))"
time="2019-07-13T01:03:28.684492128Z" level=error msg="Failed to delete firewall mark rule in sbox lb_1rql (lb-Over): reexec failed: exit status 8"
time="2019-07-13T01:03:28.684565709Z" level=error msg="Failed add IP alias 10.0.0.2 to network 1rqlilvhciazip8c26o9hf3dr LB endpoint interface eth0: cannot assign requested address"
```

Full logs attached:
[docker.log](https://github.com/moby/moby/files/3388308/docker.log)

--- a/cmd/dockerd/daemon.go
+++ b/cmd/dockerd/daemon.go
@@ -611,11 +611,17 @@ func newAPIServerConfig(cli *DaemonCli) (*apiserver.Config, error) {
 
 func loadListeners(cli *DaemonCli, serverConfig *apiserver.Config) ([]string, error) {
 	var hosts []string
+	seen := make(map[string]struct{}, len(cli.Config.Hosts))
+
 	for i := 0; i < len(cli.Config.Hosts); i++ {
 		var err error
 		if cli.Config.Hosts[i], err = dopts.ParseHost(cli.Config.TLS, honorXDG, cli.Config.Hosts[i]); err != nil {
 			return nil, errors.Wrapf(err, "error parsing -H %s", cli.Config.Hosts[i])
 		}
+		if _, ok := seen[cli.Config.Hosts[i]]; ok {
+			continue
+		}
+		seen[cli.Config.Hosts[i]] = struct{}{}
 
 		protoAddr := cli.Config.Hosts[i]
 		protoAddrParts := strings.SplitN(protoAddr, "://", 2)

